### PR TITLE
[tcl] Actually enable option utfmax when building with feature utfmax

### PR DESCRIPTION
**Pull request by shrinny**
Adds utfmax to `TCL_BUILD_OPTS` instead of time64bit (which isn't even an option in TCL's build anymore).

### CODE REVIEW

**Comment by ras0219-msft** [comment]
Blocked on #12254 so we can ensure the `vcpkg_build_nmake()` changes do not affect any other ports.

**Comment by ras0219-msft** [comment]
(self TODO: we aren't hashing `vcpkg_build_nmake()` into `vcpkg_install_nmake()`, so these changes aren't properly tested)

**Review by JackBoosY — CHANGES_REQUESTED** [review]
Please bump the port version. See [documentation](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#update-the-version-field-in-the-control-file-of-any-modified-ports).

File: scripts/cmake/vcpkg_build_nmake.cmake [hunk 1]
```diff
@@ -134,7 +134,7 @@ function(vcpkg_build_nmake)
                 # Add install command and arguments
                 set(MAKE_OPTS ${MAKE_OPTS_BASE})
                 if (_bn_ENABLE_INSTALL)
-                    set(INSTALL_OPTS ${INSTALL_OPTS_BASE} INSTALLDIR=${CURRENT_PACKAGES_DIR}/debug)
```
Comment (JackBoosY) [inline scripts/cmake/vcpkg_build_nmake.cmake:137]: Why delete this?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by JackBoosY** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

**Pull request closed without merging**